client/src/components/Focusedzone/Focusedzone.js

```diff
@@ -0,0 +1,84 @@
+/* global document */
+import React, { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+/**
+ * Wrapper component that fires a handler (`onClickOut`) when a user clicks
+ * outside of the component.
+ *
+ *
+ * Example:
+ *
+ * ```jsx
+ *    class TestComponent extends React.Component {
+ *      ...
+ *
+ *      handleClickOut() {
+ *        alert('User clicked outside of TestComponent.');
+ *      }
+ *
+ *      render() {
+ *        return (
+ *          <Focusedzone onClickOut={this.handleClickOut}>
+ *            <div>... </div>
+ *          </Focusedzone>
+ *        );
+ *       }
+ *    < /Focusedzone>
+ *  ```
+ */
+
+class Focusedzone extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.wasClicked = false;
+    this.handleElementClick = this.handleElementClick.bind(this);
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+  }
+
+  componentDidMount() {
+    const el = ReactDOM.findDOMNode(this);
+
+    // Click or touch event handlers on `el` and `document` will be both
+    // triggered when user clicked or touched inside the compoent's DOM. The
+    // handler on `el` is triggered before the one on `document` based on event
+    // bubbling.
+    ['click', 'touchstart'].forEach((eventName) => {
```

Comment:
I'm a bit skeptical using the touch events, they almost always lead to undesired UX problems...
e.g. `touchstart` triggers long before a `click` event and they can trigger together. `touchstart` can also cause closing the modal when you accidentally try scrolling the screen by using the backdrop edge of the modal, etc...

Might be worth breaking this out to another story and leave this as `click` only for this work. Investigate a polyfill to handle touch events maybe?
https://github.com/jquery/PEP

PS: forgot to remove `touchstart` events below on unmount ;)